src/emucore/Event.hxx

```diff
@@ -62,6 +62,9 @@ class Event
       RightKeyboard7, RightKeyboard8, RightKeyboard9,
       RightKeyboardStar, RightKeyboard0, RightKeyboardPound,
 
+      LeftDrivingCCW, LeftDrivingCW, LeftDrivingFire, LeftDrivingAnalog,
```

Comment:
Do we really need the new events? That way people have to remap twice (joystick and driving) when e.g. changing/adding a fire button mapping. Couldn't the digital evens be defined by the joystick events (like now) and the analog events come from the analog paddle events? 
IMO that would reduce complexity. But I am not sure how well this would work with the controller you are targeting these changes for.

Reply:
I think the problem I was trying to overcome here is that the best mapping for the classic controller is to unmap the digital events, and only map the analogue events, when driving. It's essentially an original Atari-style joystick controller (8 directional) whose shaft rotates to provide an analogue paddle equivalent. The problem I ran into when leaving digital controls enabled is in the heat of the moment, it's quite hard to avoid also triggering them when you only intended to twist.

Reply:
Got you. So this affects the CW/CCW events only, right? I wonder if some kind of automatic unmapping/ignoring of digital evens when analog events are detected would be the better option. So when digital input is detected, analog input is disabled and vice versa. Would that work for you too?

Reply:
It could work I think. For our use case, I think it would ideally be an upfront decision to use digital versus analog controls, that doesn't get re-evaluated while the game is playing - otherwise we'd run into the same issue with perhaps the analog controls being overridden accidentally. This might be even worse/more confusing than the old behaviour if there was hysteresis on the decision. 

I noticed that stella doesn't currently permit digital and analogue input simultaneously for paddles, but they are separate controls (but maybe more ambiguous per game, because the paddle might control something vertically oriented rather than horizontally). It seems like a similar solution might work for both driving and paddles though. 

I'm not sure what this solution would look like in practice? Detect an analogue control mapped for the paddles, perhaps, and ignore any digital controls mapped the joystick when driving?

Reply:
My idea was much simpler: Automatically disable the other control option during a session. So when you load a driving game and the **first** input is analog, digital input will be disabled. And vice versa. What do you think?

You can control paddle games with e.g. cursor keys, analog joystick or real paddles in parallel. It just doesn't work well, because paddle input is absolute and will overwrite any relative input resulting from mouse movement or cursor keys.

Reply:
I experimented with this this morning. I can see the appeal, and it works reasonably in practice for me, but I'm a bit concerned about the user experience for our use case. One issue is that especially for younger users, it might be quite difficult to always avoid triggering digital inputs when manipulating the controller, causing sporadic "failures" for them where the turn controls don't work. I think we'd rather the controls were unambiguous too; it'd be better if we could rely on always rotating the joystick during driving games, without leaving open the possibility of digital control to muddy the waters. One big advantage of this is that users can reasonably (and predictably) be expected to discover the controls. We're not expecting the majority of users to customise their input configuration, or even see the options dialogs.

Reply:
> One issue is that especially for younger users, it might be quite difficult to always avoid triggering digital inputs when manipulating the controller, causing sporadic "failures" for them where the turn controls don't work. 

You mean initially? Because later on this doesn't matter.

> I think we'd rather the controls were unambiguous too; it'd be better if we could rely on always rotating the joystick during driving games, without leaving open the possibility of digital control to muddy the waters. One big advantage of this is that users can reasonably (and predictably) be expected to discover the controls. 

I don't have your Atari VCs controller available, so I cannot assume anything here. But for other controllers, I don't see a problem here. I suppose you see a risk for the VCS controller, right?

> We're not expecting the majority of users to customise their input configuration, or even see the options dialogs.

I agree here. And maybe that's the decisive argument. While your solution still increases the code complexity, it doesn't necessarily do the same for the vast majority of Stella users. IMO the increased code complexity isn't a major issue. 

OK, let's keep the separate mapping. Are you going to update documentation too? Else I can do that.

Reply:
I've had a go at this in #868.